src/node_resource_limits.cc

```diff
@@ -0,0 +1,72 @@
+#include "node_resource_limits.h"
+
+#include <fstream>
+
+#ifdef __POSIX__
+#include <sys/stat.h>
+#endif  // __POSIX__
+
+namespace node {
+namespace resource_limits {
+
+#ifdef __POSIX__
+bool CgroupsResourceLimits::CgroupsAvailable(std::string cgroups_directory) {
+  struct stat stat_buf;
+  int err = stat(cgroups_directory.c_str(), &stat_buf);
+  if (err) {
+    return false;
+  } else {
+    return !!(stat_buf.st_mode & S_IFDIR);
+  }
+}
+
+bool CgroupsResourceLimits::MemParse(uint64_t* output, std::string input) {
+  int64_t value = std::atoll(input.c_str());
+  if (value <= 0 && input != "0") {  // -1 is a valid value
+    return false;
+  }
+  char suffix = input.at(input.length() - 1);
+  uint64_t multiplier = 1;
+  switch (suffix) {
+    case 'g':
+    case 'G':
+      multiplier <<= 10;
+    case 'm':
+    case 'M':
+      multiplier <<= 10;
+    case 'k':
+    case 'K':
+      multiplier <<= 10;
+  }
+  *output = static_cast<uint64_t>(value) * multiplier;
+  return true;
+}
+
+bool CgroupsResourceLimits::GetParamAsBytes(
+    uint64_t* output, std::string controller, std::string param) {
+  std::string file_contents;
+  std::ifstream file;
+  file.open(cgroups_directory_ + "/" + controller + "/" + param);
+  std::getline(file, file_contents);
+  file.close();
+  return CgroupsResourceLimits::MemParse(output, file_contents);
+}
+#endif  // __POSIX__
+
+void InitializeResourceLimits() {
+  per_process::resource_limits = std::make_unique<ResourceLimits>();
+#ifdef __POSIX__
+  if (CgroupsResourceLimits::CgroupsAvailable()) {
```

Comment:
If procfs isn't mounted when node is started but then is mounted later on, you want node to pick that up next time an isolate is created. This check prevents that.